Aspire Budgeting/Views/FileSelectorView.swift

```diff
@@ -2,57 +2,67 @@
 //  FileSelectorView.swift
 //  Aspire Budgeting
 //
-//  Created by TeraMo Labs on 10/29/19.
-//  Copyright © 2019 TeraMo Labs. All rights reserved.
-//
 
 import GoogleSignIn
 import SwiftUI
 
 struct FileSelectorView: View {
-  @EnvironmentObject var userManager: UserManager<GIDGoogleUser>
-  @EnvironmentObject var driveManager: GoogleDriveManager
-  @EnvironmentObject var sheetsManager: GoogleSheetsManager
+  let viewModel: FileSelectorViewModel
+
+  var files: [File] {
+    viewModel.getFiles() ?? [File]()
+  }
 
-  @State var selectedFile: File?
+  var error: Error? {
+    viewModel.getError()
+  }
+
+  @State private var searchText = ""
 
   var body: some View {
-    ZStack {
-      if self.userManager.user != nil {
-        ZStack {
-          if self.selectedFile == nil {
-            NavigationView {
-              List(self.driveManager.fileList) { file in
-                Button(
-                  action: {
-                    self.sheetsManager.defaultFile = file
-                    self.selectedFile = file
-                  }, label: {
-                    Text(file.name)
-                  }
-                )
-              }
-              .navigationBarTitle("Link your Aspire sheet")
-            }.onAppear {
-              if self.driveManager.fileList.isEmpty {
-                self.driveManager.getFileList()
+    if viewModel.currentState == .isLoading {
+      Text("Loading")
+    }
+
+    if viewModel.currentState == .filesRetrieved {
+      NavigationView {
+        VStack {
+          SearchBar(text: $searchText)
+          List(files.filter({searchText.isEmpty ? true : $0.name.contains(searchText)})) { file in
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/aspirebudgetingmobile/aspirebudgeting_ios/blob/ff44934e95fcfbf9892d0e49040b4320aa9b4509/Aspire Budgeting/Views/FileSelectorView.swift#L31"></span>Trailing closure syntax should be used whenever possible.<br />
<code>trailing_closure</code> <code>FileSelectorView.swift:31</code></td>
    </tr>
  </tbody>
</table>